app/src/main/java/ch/epfl/sweng/eventmanager/repository/EventRepository.java

```diff
@@ -2,62 +2,67 @@
 
 import android.arch.lifecycle.LiveData;
 import android.arch.lifecycle.MutableLiveData;
+import android.support.annotation.NonNull;
 import ch.epfl.sweng.eventmanager.repository.data.Event;
-import ch.epfl.sweng.eventmanager.repository.data.EventOrganizer;
+import com.google.firebase.database.*;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
 
 /**
  * @author Louis Vialar
  */
 @Singleton
 public class EventRepository {
-    // In the future, the events data will come from Firebase
-    // From now, we just get them from a static list
-    private final List<Event> EVENTS = new ArrayList<>(3);
-
-    {
-        EVENTS.add(
-                new Event(1, "Japan Impact", "La plus grande convention sur la culture japonaise " +
-                        "de Suisse Romande !",
-                        new EventOrganizer(1, "PolyJapan", "La commission de l'AGEPoly qui promeut la culture japonaise sur le campus et ses environs", null),
-                        null));
-        EVENTS.add(
-                new Event(2, "Sysmic", "Le festival de musique de l'association des étudiants de Microtechnique",
-                        new EventOrganizer(2, "Sysmic", "L'association des étudiants de Microtechnique", null),
-                        null));
-        EVENTS.add(
-                new Event(3, "Souper de section", "Retrouve tes professeurs et camarades lors du souper de section",
-                        new EventOrganizer(3, "CLIC", "L'association des étudiants en IC", null),
-                        null));
-    }
 
     @Inject
-    public EventRepository() {}
+    public EventRepository() {
+    }
 
     public LiveData<List<Event>> getEvents() {
         final MutableLiveData<List<Event>> data = new MutableLiveData<>();
+        DatabaseReference dbRef = FirebaseDatabase.getInstance().getReference("events");
 
-        data.setValue(Collections.unmodifiableList(EVENTS));
+        dbRef.addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                GenericTypeIndicator<List<Event>> typeToken = new GenericTypeIndicator<List<Event>>() {
```

Comment:
Why do we need a `GenericTypeIndicator` here ?

Reply:
Because List<Event>.class doesn't exist, and we explicitly want to retrieve a List of Events, and not just a List of objects

Reply:
Thank you for the details.